compute_endpoint/tests/unit/test_engines.py

```diff
@@ -73,6 +73,36 @@ def test_engine_submit(engine_type: GlobusComputeEngineBase, engine_runner):
     assert future.result(timeout=5) == param * 2
 
 
+@pytest.mark.parametrize(
+    "engine_type", (ProcessPoolEngine, ThreadPoolEngine, GlobusComputeEngine)
+)
+def test_engine_working_dir(
+    engine_type: GlobusComputeEngineBase,
+    engine_runner,
+    ez_pack_task,
+    serde,
+    task_uuid,
+):
+    """working dir remains constant across multiple fn invocations
+    This test requires submitting the task payload so that the execute_task
+    wrapper is used which switches into the working_dir, which created
+    working_dir nesting when relative paths were used.
+    """
+    engine = engine_runner(engine_type)
+
+    task_bytes = ez_pack_task(get_cwd)
+    future1 = engine.submit(str(task_uuid), task_bytes, resource_specification={})
+    unpacked1 = messagepack.unpack(future1.result())
+    result1 = serde.deserialize(unpacked1.data)
+
+    task_bytes = ez_pack_task(get_cwd)
+    future2 = engine.submit(str(task_uuid), task_bytes, resource_specification={})
+    unpacked2 = messagepack.unpack(future2.result())
+    result2 = serde.deserialize(unpacked2.data)
+
+    assert result1 == result2
```

Comment:
Per our conversation, this test is now serialized ( :+1: ), but I think the interesting bit is the `.submit()` calls.  This is obscured by the serde steps and by doing the task packing twice.  Consider:
```suggestion
    task_args = (str(task_uuid), ez_pack_task(get_cwd), {})

    future1 = engine.submit(*task_args)
    unpacked1 = messagepack.unpack(future1.result())  # blocks; avoid race condition

    future2 = engine.submit(*task_args)  # exact same task
    unpacked2 = messagepack.unpack(future2.result())

    assert unpacked1.data == unpacked2.data, "working dir should be idempotent"
```